request-manager/spec/parser.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020, Square, Inc.
+
+package spec
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// Parse a single request (YAML) file.
+// 'logFunc' is a Printf-like function used to log warning(s) should they occur.
+// Errors are returned, not logged.
+func ParseSpec(specFile string, logFunc func(string, ...interface{})) (Specs, error) {
+	var spec Specs
+
+	sequenceData, err := ioutil.ReadFile(specFile)
+	if err != nil {
+		return spec, err
+	}
+
+	/* Emit warning if unexpected or duplicate fields are present. */
+	/* Error if specs are incorrectly formatted or fields are of incorrect type. */
+	warn := yaml.UnmarshalStrict(sequenceData, &spec)
+	if warn != nil {
+		err = yaml.Unmarshal(sequenceData, &spec)
+		if err != nil {
+			return spec, err
+		}
+		logFunc("Warning: %s", warn)
```

Comment:
Can you log the file name here? Otherwise you can't tell which file a warning is coming from.